plugins/module_utils/foreman_helper.py

```diff
@@ -207,6 +207,9 @@ def update_resource(self, resource, old_entity, entity_dict, check_missing, chec
                         new_value = type(value)(new_value)
                 value = self._flatten_value(value)
                 new_value = self._flatten_value(new_value)
+                if type_eval:
+                    if yaml.load(new_value, Loader=yaml.FullLoader) == value:
```

Comment:
can you explain a bit why this `type_eval` branch is required? I am guessing that `value` as coming from the API can be a boolean, an int, a string, a list etc, but `new_value` is always a string? I would have expected Ansible not being stupid here, but who knows :)

Reply:
Yes, exactly it's about that, but it not working in any case. We are working on something more specific. And will probably also add a comment.

Reply:
My comment is: ewwww ;-)
Awaiting a tale about your endeavors!